### PR TITLE
Undeprecate InputNode

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

InputNode was deprecated in https://github.com/ray-project/ray/pull/40290, but it will continue to be used by APIs other than Serve, so we need to undeprecate it.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/dag/input_node.py [hunk 1]
```diff
@@ -3,18 +3,10 @@
 from ray.dag import DAGNode
 from ray.dag.format_utils import get_dag_node_str
 from ray.experimental.gradio_utils import type_to_string
-from ray.util.annotations import Deprecated
 
 IN_CONTEXT_MANAGER = "__in_context_manager__"
 
 
-@Deprecated(
```
Comment (GeneDer) [inline python/ray/dag/input_node.py:11]: Non-blocker: Should we put back `DeveloperAPI` or `PublicAPI` on those?

**Review by GeneDer — APPROVED** [review]
LGTM

File: python/ray/dag/input_node.py (hunk 1 above)
Comment (zcin) [reply python/ray/dag/input_node.py:11]: Yup thanks for the catch, added back @DeveloperAPI

**Comment by zcin** [comment]
@architkulkarni This PR is ready to merge!

**Pull request merged**